src/google/protobuf/compiler/main.cc

```diff
@@ -101,6 +105,30 @@ int ProtobufMain(int argc, char* argv[]) {
 }  // namespace protobuf
 }  // namespace google
 
-int main(int argc, char* argv[]) {
+#ifdef _MSC_VER
+std::string ToMultiByteUTF8String(const wchar_t* input)
+{
+    int size = WideCharToMultiByte(CP_UTF8, 0, input, wcslen(input), 0, 0, NULL, NULL);
+    std::string result(size, 0);
+    if (size) WideCharToMultiByte(CP_UTF8, 0, input, wcslen(input), &result[0], size, NULL, NULL);
+    return result;
+}
+
+int main(int argc, char* argv[]) 
+{
+  wchar_t** wargv = CommandLineToArgvW(GetCommandLineW(), &argc);
+  std::string* argv_mbcs_str = new std::string [argc];
```

Comment:
you can drop this array and leak each string inside the loop (since we leak all the strings anyway)

Reply:
Do you want me to change to leak so its more simple, or do you think its fine?

Reply:
I would prefer you drop this array to keep the code a bit more simple

Reply:
Fixed in [8e9879b](https://github.com/protocolbuffers/protobuf/pull/14253/commits/8e9879bacfeecedbb5bfee6150d580b89fd55c4e)
Could not test on my windows machine currently, will this be validated by any tests?